spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/ModuleCommands.java

```diff
@@ -191,14 +195,45 @@ public Object getValue(int row, int column) {
 				int currentRow = row - 1;
 				if (mappings.get(key).size() > currentRow) {
 					return mappings.get(key).get(currentRow);
-				} else {
+				}
+				else {
 					return null;
 				}
 			}
 		};
 		return DataFlowTables.applyStyle(new TableBuilder(model)).build();
 	}
 
+	@CliCommand(value = IMPORT_APPS, help = "Register all apps listed in a properties file")
+	public String importFromResource(
+			@CliOption(mandatory = true,
+					key = {"", "uri"},
+					help = "URI for the properties file")
+			String uri,
+			@CliOption(key = "force",
+				help = "force update if any app already exists (only if not in use)",
+				specifiedDefaultValue = "true",
+				unspecifiedDefaultValue = "false")
+			boolean force) {
+		moduleOperations().importFromResource(uri, force);
+		return String.format("Successfully registered apps from '%s'", uri);
+	}
+
+	@CliCommand(value = REGISTER_APPS, help = "Register all apps provided in a map")
+	public String registerAll(
+			@CliOption(mandatory = true,
+					key = {"", "map"},
+					help = "apps as key/value pairs with key as 'type.name' and value as a URI")
+			Properties apps,
```

Comment:
@ericbottard: what do you think of this plan (at least for M3)... we could support both `--properties` and `--propertiesFile`, but we could use comma as a delimiter in the case of inline `--properties` since that is at least consistent with deployment properties. My only concern is whether "properties" is intuitive at all when someone is providing apps via key/value pairs.